File: src/Engine/Managers/EnvironmentManager.java
package Engine.Managers;

import Engine.Engine;

/**
 * Created by steppers on 03/11/15.
 */
public class EnvironmentManager extends Manager {

    private static EnvironmentManager instance = new EnvironmentManager();
    public static EnvironmentManager getInstance(){return instance;}

    private Engine engine;

    public void init(Engine engine)
    {
        this.engine = engine;
    }

    public void stopExecution()
    {
        engine.stop();
    }

}

File: src/Engine/Managers/EnvironmentManager.cpp
//
// Created by Ollie on 05/11/2015.
//

#include "EnvironmentManager.h"

void EnvironmentManager::init(Engine* engine) {
    _engine = engine;
}

void EnvironmentManager::stopEngine() {
    _engine->stop();
}

void EnvironmentManager::setDebugMode(bool enabled) {
    _DEBUG_MODE = enabled;
}

bool EnvironmentManager::getDebugMode() {
    return _DEBUG_MODE;
}